pennylane_lightning/_serialize.py

```diff
@@ -87,6 +87,8 @@ def _serialize_named_obs(ob, wires_map: dict, use_csingle: bool):
     """Serializes a Named observable"""
     named_obs = NamedObsC64 if use_csingle else NamedObsC128
     wires = [wires_map[w] for w in ob.wires]
+    if ob.name == "Identity":
+        wires = wires[:1]
```

Comment:
new arithmetic operators might decompose into a new Identity with multiple wires, but the C++ hard-codes Identity to only have 1 wire. @mlxd I remember there was some concern about this new Identity feature - is this a bad thing to do here? It feels not great.

Reply:
ok I at least figured out why I only sometimes saw the issue. In non-adjoint expectation measurements, this is [handled by QubitDevice](https://github.com/PennyLaneAI/pennylane-lightning/blob/08dfbe0f8e7f962f2034b032e63ad2fa4eed0e1e/pennylane_lightning/lightning_qubit.py#L803), thereby never calling `_serialize_ob(servables)` on it. However, adjoint-diff QNodes don't do this check while running `_process_jacobian_tape`, and will [blindly call](https://github.com/PennyLaneAI/pennylane-lightning/blob/08dfbe0f8e7f962f2034b032e63ad2fa4eed0e1e/pennylane_lightning/lightning_qubit.py#L486) `_serialize_observables` on it. Is there a strong reason we do the pre-processing and defaulting to super in only non-adjoint computations?

Reply:
Does this solve the identity situation as mentioned earlier? Does stripping the wires remove Identities as @chaeyeunpark mentioned?

Reply:
Thanks @timmysilv and fine by me. You ok with this @chaeyeunpark ?

Reply:
Yeap, it looks good.